llvm/lib/Transforms/Instrumentation/MemorySanitizer.cpp

```diff
@@ -4557,7 +4561,13 @@ struct MemorySanitizerVisitor : public InstVisitor<MemorySanitizerVisitor> {
       return;
     Value *SizeVal =
       IRB.CreateTypeSize(MS.IntptrTy, DL.getTypeStoreSize(ElemTy));
-    IRB.CreateCall(MS.MsanInstrumentAsmStoreFn, {Operand, SizeVal});
+    if (MS.CompileKernel) {
+      IRB.CreateCall(MS.MsanInstrumentAsmStoreFn, {Operand, SizeVal});
+    } else {
+      auto [ShadowPtr, _] =
+          getShadowOriginPtrUserspace(Operand, IRB, IRB.getInt8Ty(), Align(1));
+      IRB.CreateAlignedStore(getCleanShadow(ElemTy), ShadowPtr, Align(1));
```

Comment:
Any reason we use the the Aligned variant? Align(1) should be the same as unspecified alignment, so we can just use `CreateStore`, right?

Reply:
ElemTy` utilizes `elementtype` (`ptr elementtype(i32) @id1` => i32, which does not encode the alignment).
An unspecified alignment uses the default, `ElemTy`'s alignment, which is 4 in the i32 case.
However, if the element type is actually unaligned, `CreateStore` could cause a shadow misalignment issue on certain architectures using `-mstrict-align`.

The tests specifically write `, align 1`

Reply:
Can you add this as a comment? This is quite subtle and very likely lost on cursory readers.

Reply:
Done:)